server/controllers/ticket/close.php

```diff
@@ -62,10 +81,14 @@ public function handler() {
     }
 
     private function shouldDenyPermission() {
-        $user = Controller::getLoggedUser();
+        if(Controller::isUserSystemEnabled() || Controller::isStaffLogged()) {
+            $user = Controller::getLoggedUser();
 
-        return (!Controller::isStaffLogged() && $this->ticket->author->id !== $user->id) ||
-               (Controller::isStaffLogged() && $this->ticket->owner && $this->ticket->owner->id !== $user->id);
+            return (!Controller::isStaffLogged() && $this->ticket->author->id !== $user->id) ||
+                   (Controller::isStaffLogged() && $this->ticket->owner && $this->ticket->owner->id !== $user->id);
+        } else {
+            return $this->ticket->ticket_number != Session::getInstance()->getTicketNumber();
```

Comment:
Isn't this already checked in line 50 by the validation?

I think is not documented, but if the validations (configured by the array that the "validations" function returns) don't pass, the handler function does not execute.